tests/numpy_eval_test.py

```diff
@@ -0,0 +1,62 @@
+# Copyright 2018 Google LLC
```

Comment:
I wonder if it would be better to systematically add this kind of testing to `lax_test.py`. We already have `CompileAndCheck`, which compares op-by-op vs JIT evaluations of a primitive. Perhaps we should simply add the NumPy evaluator as a third option?

This might also make many of the CheckAgainstNumpy calls in that file redundant, since they don't actually check against NumPy, rather `lax_reference`.

Reply:
@hawkinsp I already do exactly what you describe (see the diff of `lax_test.py`). numpy_eval_test only does some extra tests, i. e. check `numpy_eval` works on device array etc.

It's definitely tempting to remove `lax_reference.py` by replacing all usages with `numpy_eval()(lax.some_method)`. However, this would degrade tests for lax functions that are more than just a wrapper around a primitive, i. e. `lax.acos`. We want to compare to `np.acos` here, not the `numpy_eval()(lax.acos)`! If `lax.acos` had a bug, we wouldn't catch it otherwise. There are some more of these examples.

I think the best thing is to compare to both, as currently implemented.